spirv_msl.cpp

```diff
@@ -3097,6 +3177,32 @@ void CompilerMSL::emit_store_statement(uint32_t lhs_expression, uint32_t rhs_exp
 	}
 }
 
+// Metal expands float[]/float2[] members inside structs to float4[] so we must unpack
+string CompilerMSL::to_dereferenced_expression(uint32_t id, bool register_expression_read)
```

Comment:
Is it necessary to create an override here? Handling these cases should already work without a redundant override.

Reply:
Removed this part. Seems to work without this change.